ceasiompy/CPACS2GMSH/tests/test_advancemeshing.py

```diff
@@ -137,6 +144,174 @@ def test_restrict_fields():
     gmsh.finalize()
 
 
+def test_min_fields():
+    """
+    Test if a simple min field can be generate for a restrict field
+    """
+    gmsh.initialize()
+    # create a sphere with gmsh
+
+    gmsh.model.occ.addSphere(0, 0, 0, 1)
+    # Create a distance fields on the first line of the sphere
+    mesh_fields = {"nbfields": 0, "restrict_fields": []}
+
+    mesh_fields = distance_field(mesh_fields, 1, [1])
+
+    # create a simple matheval field on the previous distance_field
+    mesh_fields["nbfields"] += 1
+    gmsh.model.mesh.field.add("MathEval", mesh_fields["nbfields"])
+    gmsh.model.mesh.field.setString(mesh_fields["nbfields"], "F", "(F1 /2)")
+    # Create a restrict field on one of the sphere surface
+    mesh_fields = restrict_fields(mesh_fields, 2, [1])
+
+    # create the min field
+    mesh_fields = min_fields(mesh_fields)
+
+    # Check mesh field are created
+    assert mesh_fields["nbfields"] == 4
+
+    # Check mesh field tags are correct
+
+    assert gmsh.model.mesh.field.getType(1) == "Distance"
+    assert gmsh.model.mesh.field.getType(2) == "MathEval"
+    assert gmsh.model.mesh.field.getType(3) == "Restrict"
+    assert gmsh.model.mesh.field.getType(4) == "Min"
+
+    gmsh.clear()
+    gmsh.finalize()
+
+
+@pytest.mark.skipif(
+    sys.platform == "darwin", reason="'synchronize' function causes segmentation fault on macOS"
+)
+def test_compute_area():
+    """
+    Test if the area of a surface is correctly computed
+    """
+    gmsh.initialize()
+
+    # create a square with gmsh
+
+    square = gmsh.model.occ.addRectangle(0, 0, 0, 1, 1)
+    gmsh.model.occ.synchronize()
+    gmsh.model.mesh.generate(2)
+
+    # compute the meshed area
+
+    area = compute_area(square)
+
+    # Check the area is correct
+
+    assert pytest.approx(area, 0.01) == 1
+
+    gmsh.clear()
+    gmsh.finalize()
+
+
+@pytest.mark.skipif(
+    sys.platform == "darwin", reason="'synchronize' function causes segmentation fault on macOS"
+)
+def test_refine_wing_section():
+    """
+    Test if the wing section is correctly refined by the advancemeshing algorithm
+    """
+
+    CPACS_IN_PATH = Path(CPACS_FILES_PATH, "simple_untruncated.xml")
+    cpacs = CPACS(str(CPACS_IN_PATH))
+
+    export_brep(cpacs, TEST_OUT_PATH)
+
+    generate_gmsh(
+        CPACS_IN_PATH,
+        TEST_OUT_PATH,
+        TEST_OUT_PATH,
```

Comment:
Use only one input for the result file and create the "brep_file" dir in the functions.
Should be change everywhere in the code.